ygen/goelements.go

```diff
@@ -538,6 +553,50 @@ func (s *GoLangMapper) goUnionSubTypes(subtype *yang.YangType, ctx *yang.Entry,
 	return errs
 }
 
+// generateGoDefaultValue returns a pointer to a Go literal that represents the
+// default value for the entry. If there is no default value for the field, nil
+// is returned.
+func generateGoDefaultValue(field *yang.Entry, mtype *MappedType, gogen *GoLangMapper, compressPaths, skipEnumDedup, shortenEnumLeafNames, useDefiningModuleForTypedefEnumNames bool, enumOrgPrefixesToTrim []string, simpleUnions bool) (*string, error) {
+	// Set the default type to the mapped Go type.
+	defaultValues := field.DefaultValues()
+	if len(defaultValues) == 0 && mtype.DefaultValue != nil {
+		defaultValues = []string{*mtype.DefaultValue}
+	}
+	for i, defVal := range defaultValues {
+		var err error
+		if defaultValues[i], _, err = gogen.yangDefaultValueToGo(defVal, resolveTypeArgs{yangType: field.Type, contextEntry: field}, len(mtype.UnionTypes) == 1, compressPaths, skipEnumDedup, shortenEnumLeafNames, useDefiningModuleForTypedefEnumNames, enumOrgPrefixesToTrim); err != nil {
+			return nil, err
+		}
+	}
+	// TODO(wenbli): In ygot v1, we should no longer
```

Comment:
This TODO makes me feel like we could deprecate this now -- but I'm open to discussing this again.

Reply:
As above I'm wondering if we should deprecate all at once at v1 release.